book.rb

```diff
@@ -0,0 +1,13 @@
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(date, person)
+    @rentals.push(Rental.new(date, self, person)) unless @rentals.include?(Rental.new(date, self, person))
```

Comment:
- [ ] Hey! Let's make a thoughtful adjustment to this implementation. In your rental class, you're already appending the newly created book to the rentals array. 📚✨ No need to push it again in the add_rental method, as it could lead to duplication. 🚫🔄 This ensures a cleaner and more efficient way of managing rentals. Keep up the good work! 👍🏽📖

```ruby
 def add_rental(person, date)
    Rental.new(date, self, person)
  end
```